redisvl/providers/huggingface.py

```diff
@@ -10,16 +9,23 @@ def __init__(self, model: str, api_config: Optional[Dict] = None):
         # TODO set dims based on model
         dims = 768
         super().__init__(model, dims, api_config)
+        try:
+            from sentence_transformers import SentenceTransformer
+        except ImportError:
+            raise ImportError(
+                "Huggingface provider requires sentence-transformers library. Please install with pip install sentence-transformers"
+            )
+
         self._model_client = SentenceTransformer(model)
 
-    def embed(self, emb_input: str, preprocess: callable = None) -> List[float]:
+    def embed(self, emb_input: str, preprocess: Optional[Callable] = None) -> List[float]:
         if preprocess:
             emb_input = preprocess(emb_input)
         embedding = self._model_client.encode([emb_input])[0]
         return embedding.tolist()
 
     def embed_many(
-        self, inputs: List[str], preprocess: callable = None, chunk_size: int = 1000
+        self, inputs: List[str], preprocess: Optional[Callable] = None, chunk_size: int = 1000
```

Comment:
this notebook from google shows how to use tenacity for retry and exponential backoff to deal with rate limits: https://github.com/GoogleCloudPlatform/generative-ai/blob/8e19bc6a5de9f253444a39f82790046b1817f89d/language/examples/document-qa/question_answering_large_documents.ipynb

Reply:
Going to make this a followup issue